runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDos.java

```diff
@@ -146,6 +148,20 @@ public static ParDoPayload toProto(ParDo.MultiOutput<?, ?> parDo, SdkComponents
     return doFnAndMainOutputTagFromProto(payload.getDoFn()).getMainOutputTag();
   }
 
+  public static RunnerApi.PCollection getMainInput(
```

Comment:
This seems unused except in test. Have I missed something?

Should javadoc the requirement that it is a ParDo, or just decompose the input outside the method, and perhaps also make it private.